client/js/components/CheckAuth.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import decodeToken from '../utils/decodeToken';
+import logout from '../actions/logoutUser';
+
+const AuthenticateRoutes = (component) => {
+  /**
+   *
+   *
+   * @class CheckAuth
+   * @extends {React.Component}
+   */
+  class CheckAuth extends React.Component {
+    /**
+   * Creates an instance of CheckAuth.
+   * @param {any} props
```

Comment:
Please include the type of parameter and also a description for all your documentation, e.g

`@param {object} props - description of props`